nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -153,23 +158,53 @@ protected void captureAndVerifyAnnotationEventNotification(Supplier<DocumentMode
             assertEquals(1, handledEvents.size());
             Event expectedEvent = handledEvents.get(0);
             assertEquals(annotationEventType, expectedEvent.getName());
-            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel);
-            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel,
-                    handledEvents.get(0), annotationEventType);
+
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel,
+                    annotatedDocumentModel);
+
+            // Deduct only the emails sent since the last time
+            List<SmtpMailServerFeature.MailMessage> previousMails = new ArrayList<>(mails);
+            mails.clear();
+            mails.addAll(emailsResult.getMails()
+                                     .stream()
+                                     .filter(Predicate.not(previousMails::contains))
+                                     .collect(Collectors.toList()));
```

Comment:
As the tests should be independent, you should not need to do this weird filtering.

Reply:
They still are, but on a per test basis every `emailsResult.getMails()` will get all the mails since the begining of the test and during one test with a reply you will first receive the notification for the first comment (same event).
At first the test was written without the filtering but it was complexifying the internal methods (this one especially) and their reuse as the mail count is checked...
In https://github.com/nuxeo/nuxeo/pull/3663/commits/c0ca914e28e03369b0e7c3a364a85237a3690131#diff-b5ba4c2a86feb19034eeccb42bbcd8aaR171 
I needed to bypass the mailcount check which only varies between 0 and 1 so I introduced an offset.

Reply:
Link to https://github.com/nuxeo/nuxeo/pull/3587#discussion_r367480270, just do not subscribe to the notifications when initializing stuff for your test, then subscribe, do the annotation, assert the email.